### DISCUSSION TITLE
RBAC Oauth2 Google

### DISCUSSION THREAD

**Discussion by fedeostrit** [provectus/kafka-ui#3274] [Q&A] [answered]
Hi, I wanted to know if there is any way to configure google's oauth2 and use the google provider to do RBAC by groups and not just by user or entire domain as shown in the documentation:
oauth2:

kafka:
  clusters:
    - name: local
      bootstrapServers: localhost:9092
    # ...

auth:
  type: OAUTH2
  oauth2:
    client:
      google:
        provider: google
        clientId: xxx.apps.googleusercontent.com
        clientSecret: GOCSPX-xxx
        user-name-attribute: email
        custom-params:
          type: google
          allowedDomain: provectus.com

provider:
    - name: "memelords"
      subjects:
        - provider: oauth_google
          type: domain
          value: "memelord.lol"

It only allows you to use domain or user, but if I have 100 users, I would have to create a role for each one and map it in a huge .yaml file. Also, when a new resource enters, it must be modified to add it.

**Comment by Haarolean** [comment, marked as answer]
Planned: https://github.com/provectus/kafka-ui/issues/3279
